### PR TITLE
add draft-release support

### PR DESCRIPTION
**Pull request by developer-guy**
as we merged https://github.com/slsa-framework/slsa-github-generator/pull/1882, we need to test it requested by @ianlewis and @laurentsimon in https://github.com/slsa-framework/slsa-github-generator/issues/1476. So, this PR aims to create a test for draft-release support.

### CODE REVIEW

File: .github/workflows/e2e.go.tag.main.config-ldflags-assets-draft-tag.slsa3.yml [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3 # v3.5.0
+      - id: create
+        run: |
+          set -euo pipefail
+
+          ./.github/workflows/scripts/e2e-create-release.sh
```
Comment (laurentsimon) [inline .github/workflows/e2e.go.tag.main.config-ldflags-assets-draft-tag.slsa3.yml:30]: I think you need to update this file to mark the release as a draft in the API call.

**Review by laurentsimon — COMMENTED** [review]
Thanks for the PR!

File: .github/workflows/e2e.go.tag.main.config-ldflags-assets-draft-tag.slsa3.yml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      - "v[0-9]+.[0-9]+.[0-9]+" # # triggers only if push new tag version, like `v0.8.4` or else
+
+permissions: read-all
+
+env:
+  GH_TOKEN: ${{ secrets.E2E_GO_TOKEN }}
+  ISSUE_REPOSITORY: slsa-framework/slsa-github-generator
+  # WARNING: update build job if CONFIG_FILE changes.
+  CONFIG_FILE: .github/configs-go/config-ldflags-tag.yml
+  DEFAULT_VERSION: v34.0.0
```
[the remark this answers is not in the document]
Comment (ianlewis) [reply .github/workflows/e2e.go.tag.main.config-ldflags-assets-draft-tag.slsa3.yml:16]: I added some doc on `DEFAULT_VERSION` to the [e2e tests' readme](https://github.com/slsa-framework/example-package/blob/c5f97a9746a5c47b8667aae5b88266b174193720/.github/workflows/README.md).

[the remark this answers is not in the document]
Comment (developer-guy) [reply .github/workflows/e2e.go.tag.main.config-ldflags-assets-draft-tag.slsa3.yml:16]: thanks, this is really useful!

**Comment by developer-guy** [comment]
kindly ping @laurentsimon

**Review by ianlewis — APPROVED** [review]
LGTM

**Pull request merged**